cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -186,116 +191,102 @@ func run(ctx context.Context, target, assetType, optJSON string, state checkstat
 }
 
 func processVulns(results ScanResponse, registryEnvDomain, target string, state checkstate.State) error {
-	// If there are no vulnerabilities we can return.
-	if len(results) < 1 || len(results) == 1 && len(results[0].Vulnerabilities) == 0 {
-		return nil
-	}
+	for _, tt := range results {
+
+		outdatedPackageVulns := make(map[outdatedKey][]outdatedPackage)
+		for _, tv := range tt.Vulnerabilities {
+			key := outdatedKey{
+				name:    tv.PkgName,
+				version: tv.InstalledVersion,
+				path:    tv.PkgPath,
+			}
+			l, ok := outdatedPackageVulns[key]
+			if !ok {
+				l = []outdatedPackage{}
+			}
+			outdatedPackageVulns[key] = append(l, outdatedPackage{
+				severity: tv.Severity,
+				fixedBy:  tv.FixedVersion,
+				cve:      tv.VulnerabilityID,
+				link:     tv.PrimaryURL,
+				cves:     tv.CweIDs,
+			})
+		}
 
-	outdatedPackageVulns := make(map[string]outdatedPackage)
-	for _, trivyTarget := range results {
-		for _, dockerVuln := range trivyTarget.Vulnerabilities {
-			op, ok := outdatedPackageVulns[dockerVuln.PkgName]
-			if ok {
-				if isMoreSevere(dockerVuln.Severity, op.severity) {
-					op.severity = dockerVuln.Severity
+		vp := report.ResourcesGroup{
+			Name: "Package Vulnerabilities",
+			Header: []string{
+				"FixedBy",
+				"CVE",
+				"Severity",
+				"CWEs",
+				"Link",
+			},
+		}
+		for key, l := range outdatedPackageVulns {
+
+			// Sort CVEs by severity
+			sort.Slice(l, func(i, j int) bool {
+				return getScore(l[i].severity) > getScore(l[j].severity)
+			})
+
+			vp.Rows = []map[string]string{}
+			maxScore := getScore("NONE")
+			for i, l := range l {
+				if i > vulnCVETrucateLimit {
+					break
 				}
-				if version.Compare(version.Normalize(dockerVuln.FixedVersion), version.Normalize(op.fixedBy), ">") {
-					op.fixedBy = dockerVuln.FixedVersion
+				newScore := getScore(l.severity)
+				if newScore > maxScore {
+					maxScore = newScore
 				}
-			} else {
-				op = outdatedPackage{
-					name:     dockerVuln.PkgName,
-					version:  dockerVuln.InstalledVersion,
-					severity: dockerVuln.Severity,
-					fixedBy:  dockerVuln.FixedVersion,
-					cves:     []string{},
+				row := make(map[string]string, len(vp.Header))
+				row["FixedBy"] = l.fixedBy
+				if l.cves != nil {
+					row["CWEs"] = strings.Join(l.cves, ",")
 				}
+				row["CVE"] = l.cve
+				if l.link == "" {
```

Comment:
Does it make sense to use the trivy provided link vs the nist.gov one?

- https://avd.aquasec.com/nvd/2021/cve-2021-21707/
- https://nvd.nist.gov/vuln/detail/cve-2021-21707

Reply:
I think is fair as long we are using the tool that provides the info.